run

```diff
@@ -159,7 +189,26 @@ clean() {
 restore() {
   if did restore ; then return 0 ; fi
 
-  paket --silent restore
+  if [ "$USE" = dotnet ] ; then
+    local NEW_CHECKSUM
+    if has md5 ; then
+      SUM=md5
+    elif has md5sum ; then
+      SUM=md5sum
+    else
+      echo "Couldn't find command to compute MD5."
+      return 1
+    fi
+    NEW_CHECKSUM="$(find . -name 'project.json' -print0 | xargs -0 cat | $SUM)"
+    local OLD_CHECKSUM=""
+    if [ -f .dotnet ] ; then OLD_CHECKSUM="$(cat .dotnet)" ; fi
+    if [ "$NEW_CHECKSUM" != "$OLD_CHECKSUM" ] ; then
+      echo -n "$NEW_CHECKSUM" > .dotnet
+      dotnet restore --verbosity Warning
```

Comment:
At least on my OS X machine (and I see same under VMware+Windows) `dotnet restore` takes a lot of time even when it has already been run before.  Here are two runs of `dotnet restore` with nothing in between:

```
$ USE=dotnet time dotnet restore --verbosity Warning
warn : Detected package downgrade: System.Threading.Tasks from 4.0.11 to 4.0.0 
warn :  FibonacciMultiJit (>= 1.0.0) -> BenchmarkDotNet (>= 0.9.9) -> BenchmarkDotNet.Toolchains.Roslyn (>= 0.9.9) -> Microsoft.CodeAnalysis.CSharp (>= 1.3.2) -> Microsoft.CodeAnalysis.Common (= 1.3.2) -> System.Threading.Tasks (>= 4.0.11) 
warn :  FibonacciMultiJit (>= 1.0.0) -> BenchmarkDotNet (>= 0.9.9) -> BenchmarkDotNet.Toolchains.Roslyn (>= 0.9.9) -> System.Threading.Tasks (>= 4.0.0)
       19.61 real        48.80 user         0.97 sys

$ USE=dotnet time dotnet restore --verbosity Warning
warn : Detected package downgrade: System.Threading.Tasks from 4.0.11 to 4.0.0 
warn :  FibonacciMultiJit (>= 1.0.0) -> BenchmarkDotNet (>= 0.9.9) -> BenchmarkDotNet.Toolchains.Roslyn (>= 0.9.9) -> Microsoft.CodeAnalysis.CSharp (>= 1.3.2) -> Microsoft.CodeAnalysis.Common (= 1.3.2) -> System.Threading.Tasks (>= 4.0.11) 
warn :  FibonacciMultiJit (>= 1.0.0) -> BenchmarkDotNet (>= 0.9.9) -> BenchmarkDotNet.Toolchains.Roslyn (>= 0.9.9) -> System.Threading.Tasks (>= 4.0.0)
       19.86 real        50.07 user         0.99 sys
```

The check skips `dotnet restore` unless the `package.json` files have changed:

```
$ USE=dotnet time ./run restore
warn : Detected package downgrade: System.Threading.Tasks from 4.0.11 to 4.0.0 
warn :  FibonacciMultiJit (>= 1.0.0) -> BenchmarkDotNet (>= 0.9.9) -> BenchmarkDotNet.Toolchains.Roslyn (>= 0.9.9) -> Microsoft.CodeAnalysis.CSharp (>= 1.3.2) -> Microsoft.CodeAnalysis.Common (= 1.3.2) -> System.Threading.Tasks (>= 4.0.11) 
warn :  FibonacciMultiJit (>= 1.0.0) -> BenchmarkDotNet (>= 0.9.9) -> BenchmarkDotNet.Toolchains.Roslyn (>= 0.9.9) -> System.Threading.Tasks (>= 4.0.0)
       21.00 real        52.06 user         1.08 sys

$ USE=dotnet time ./run restore
        0.02 real         0.01 user         0.01 sys
```

I use the `run` script for both incremental builds and for clean builds from scratch.  19 seconds for NOP restore is too much.  If there is way to make `dotnet restore` faster then that would be nice.